File: app/src/main/java/com/kelsos/mbrc/commands/StartDiscoveryCommand.java
package com.kelsos.mbrc.commands;

import javax.inject.Inject;
import com.kelsos.mbrc.interfaces.ICommand;
import com.kelsos.mbrc.interfaces.IEvent;
import com.kelsos.mbrc.services.ServiceDiscovery;

public class StartDiscoveryCommand implements ICommand {
  private ServiceDiscovery mDiscovery;

  @Inject public StartDiscoveryCommand(ServiceDiscovery mDiscovery) {
    this.mDiscovery = mDiscovery;
  }

  @Override public void execute(IEvent e) {
    mDiscovery.startDiscovery();
  }
}

File: app/src/main/java/com/kelsos/mbrc/commands/StartDiscoveryCommand.kt
package com.kelsos.mbrc.commands

import com.kelsos.mbrc.networking.discovery.RemoteServiceDiscovery
import com.kelsos.mbrc.networking.protocol.ProtocolAction
import com.kelsos.mbrc.networking.protocol.ProtocolMessage

class StartDiscoveryCommand(
  private val discovery: RemoteServiceDiscovery,
) : ProtocolAction {
  override fun execute(message: ProtocolMessage) {
    discovery.startDiscovery()
  }
}
